nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/GlassFishImpl.java

```diff
@@ -87,10 +87,24 @@ public synchronized void stop() throws GlassFishException {
             throw new IllegalStateException("Already in " + status + " state.");
         }
         status = Status.STOPPING;
+        sleepShutdownGracePeriod();
         gfKernel.stop();
         status = Status.STOPPED;
     }
 
+    public static void sleepShutdownGracePeriod() {
+        String shutdowngrace = System.getProperty("fish.payara.shutdowngrace");
```

Comment:
Would probably be a little cleaner to have this as a final String variable or enum rather than just specified here.

Reply:
looking into it...